ctapipe/io/tableloader.py

```diff
@@ -0,0 +1,330 @@
+            By default it is sorted by obs_id and event_id.
+        
+        Returns
+        -------
+        output: astropy.Table
+            Single table containing the enabled data information."""
+        
+        if not any(self.all_loads):
+            self.close()
+            raise ValueError("No data enabled for loading.")
+
+        if self.load_instrument and (not any([self.load_dl1_images, self.load_dl1_parameters, self.load_true_images])):
+            self.close()
+            raise ValueError("""Instrument data requires tel_id-based quantities, 
+            but only event_id-based quantities have been selected together with it.
+            No data enabled for loading.""")
+        
+        split_mode = get_structure(self.h5file)
+        
+        if labels is not None:
+            if all([isinstance(label, str) for label in labels]):
+                selected_tel_ids = []
+                for tel_type in labels:
+                    selected_tel_ids += self.subarray.get_tel_ids_for_type(tel_type)
+            elif all([isinstance(label, int) for label in labels]):
+                selected_tel_ids = labels
+            else:
+                self.close()
+                raise ValueError("Check supplied labels.")
+        
+        if split_mode == "by_id":
+
+            output = self.read_telescope_events(tel_ids=selected_tel_ids)
+            
+            try: 
+                output.sort(sort_by)
+            except ValueError:
+                output.sort(["tel_id"])
+        
+        elif split_mode == "by_type":
+            
+            output = self.read_telescope_events_by_tel_type(tel_ids = selected_tel_ids)
+        
+            try:
+                for tel_type in output.keys():
+                    output[tel_type].sort(sort_by)
+            except ValueError:
+                for tel_type in output.keys():
+                    output[tel_type].sort(["tel_id"])
+        
+        self.close()
+        
+        return output
+    
+    def read_telescope_events_by_tel_type(self, tel_ids=None):
+                
+        selected_subarray = self.subarray.select_subarray(tel_ids, name="")
+        selected_tel_types = selected_subarray.telescope_types
+        
+        table_dict = {}
+        for tel_type in selected_tel_types:
+            
+            selected_tel_ids = [tel_id for tel_id in tel_ids if tel_id in self.subarray.get_tel_ids_for_type(tel_type)]
+            table_dict[str(tel_type)] = self.read_telescope_events(selected_tel_ids)
+        
+        return table_dict
+
+    def read_telescope_events(self, tel_ids=None):
+        if tel_ids is None:
+            tel_ids = self.subarray.tel.keys()
+        
+        table = None
+        
+        # Joini all the DL1-based data, which depends on the triggered tel_ids
+        if (self.load_dl1_images or self.load_dl1_parameters or self.load_true_images) and ((table is None) or (len(table)==0)):
```

Comment:
These if / elif statements here are not necessary / much more complicated than needed since table is always `None` here, you define it in the line just above.

Reply:
removed